api/types.go

```diff
@@ -177,6 +180,81 @@ type Options struct {
 	NumThread int `json:"num_thread,omitempty"`
 }
 
+func (opts *Options) FromMap(m map[string]interface{}) error {
```

Comment:
Would it be possible to avoid reflection? mergo has a `Map` function which merges a struct into a map or vice versa.